config/environments/production.rb

```diff
@@ -130,12 +130,12 @@
   config.action_mailer.perform_deliveries = true
   config.action_mailer.default_url_options = { host: protocol + ENV["APP_DOMAIN"].to_s }
   ActionMailer::Base.smtp_settings = {
-    address: "smtp.sendgrid.net",
-    port: "587",
-    authentication: :plain,
-    user_name: "apikey",
-    password: ENV["SENDGRID_API_KEY"],
-    domain: ENV["APP_DOMAIN"],
+    address: ENV["SMTP_ADDRESS"],
+    port: ENV["SMTP_ADDRESS"],
+    authentication: ENV["SMTP_AUTHENTICATION"],
+    user_name: ENV["SMTP_USER_NAME"],
+    password: ENV["SMTP_PASSWORD"],
+    domain: ENV["SMTP_DOMAIN"] || ENV["APP_DOMAIN"],
```

Comment:
I purposely allow the domain to have a fallback to minimize the configuration we need to do. 

On a different note, Mailgun's free tier doesn't support "custom" domain" and provides you a domain name consist of a random hash. I'm not fully clear how that works and and how other SMTP providers could be different, so providing a blank slate for configuration is best.

Reply:
```suggestion
    domain: ENV["SMTP_DOMAIN"].presence || ENV["APP_DOMAIN"],
```